### PR TITLE
J22 Mizzix, Replica Rider

### PR DESCRIPTION
**Pull request by Simisays**
Thanks to Squee for the script !

Don't know why the test failed, it worked during testing 
image

### CODE REVIEW

**Comment by paulsnoops** [comment]
The test was failing just because of a time-out. I've just let it re-run, seems okay now

File: forge-gui/res/cardsfolder/upcoming/mizzix_replica_rider.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Name:Mizzix, Replica Rider
+ManaCost:4 R
+Types:Legendary Creature Goblin Wizard
+PT:4/5
+K:Flying
+T:Mode$ SpellCast | ValidCard$ Card.wasNotCastFromYourHand | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigCopy | TriggerDescription$ Whenever you cast a spell from anywhere other than your hand, you may pay {1}{U/R}. If you do, copy that spell and you may choose new targets for the copy. If the copy is a permanent spell, it gains haste and "At the beginning of the end step, sacrifice this permanent."
+SVar:TrigCopy:AB$ CopySpellAbility | Cost$ 1 UR | Defined$ TriggeredSpellAbility | MayChooseTarget$ True | RememberNewCard$ True | SubAbility$ DBAnimate
+SVar:DBAnimate:DB$ Animate | Defined$ Remembered | ConditionDefined$ Remembered | ConditionPresent$ Permanent | Keywords$ Haste | Triggers$ EOTSac | Duration$ Permanent
+SVar:EOTSac:Mode$ Phase | Phase$ End of Turn | Execute$ TrigSac | TriggerDescription$ At the beginning of the end step, sacrifice this permanent.
+SVar:TrigSac:DB$ Sacrifice | Defined$ Remembered | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/mizzix_replica_rider.txt:10]: - Defined on cards doesn't exist Sacrifice

- Cleanup happens on the wrong card

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/mizzix_replica_rider.txt]: Oh is this a recent change ? I guess just removing the defined should be fine ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/mizzix_replica_rider.txt]: I think this was an old mistake that grew over time

File: forge-gui/res/cardsfolder/upcoming/mizzix_replica_rider.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+T:Mode$ SpellCast | ValidCard$ Card.wasNotCastFromYourHand | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigCopy | TriggerDescription$ Whenever you cast a spell from anywhere other than your hand, you may pay {1}{U/R}. If you do, copy that spell and you may choose new targets for the copy. If the copy is a permanent spell, it gains haste and "At the beginning of the end step, sacrifice this permanent."
+SVar:TrigCopy:AB$ CopySpellAbility | Cost$ 1 UR | Defined$ TriggeredSpellAbility | MayChooseTarget$ True | RememberNewCard$ True | SubAbility$ DBAnimate
+SVar:DBAnimate:DB$ Animate | Defined$ Remembered | ConditionDefined$ Remembered | ConditionPresent$ Permanent | Keywords$ Haste | Triggers$ EOTSac | Duration$ Permanent | AtEOT$ Sacrifice | SubAbility$ DBCleanup
+SVar:EOTSac:Mode$ Phase | Phase$ End of Turn | Execute$ TrigSac | TriggerDescription$ At the beginning of the end step, sacrifice this permanent.
+SVar:TrigSac:DB$ Sacrifice
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+AI:RemoveDeck:Random
+DeckHas:Ability$Token|Sacrifice
+DeckHints:Ability$Token
+DeckNeeds:Type$Instant|Sorcery
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/mizzix_replica_rider.txt:15]: I think that DeckNeeds is too hard?

File: forge-gui/res/cardsfolder/upcoming/mizzix_replica_rider.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+Name:Mizzix, Replica Rider
+ManaCost:4 R
+Types:Legendary Creature Goblin Wizard
+PT:4/5
+K:Flying
+T:Mode$ SpellCast | ValidCard$ Card.wasNotCastFromYourHand | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigCopy | TriggerDescription$ Whenever you cast a spell from anywhere other than your hand, you may pay {1}{U/R}. If you do, copy that spell and you may choose new targets for the copy. If the copy is a permanent spell, it gains haste and "At the beginning of the end step, sacrifice this permanent."
+SVar:TrigCopy:AB$ CopySpellAbility | Cost$ 1 UR | Defined$ TriggeredSpellAbility | MayChooseTarget$ True | RememberNewCard$ True | SubAbility$ DBAnimate
+SVar:DBAnimate:DB$ Animate | Defined$ Remembered | ConditionDefined$ Remembered | ConditionPresent$ Permanent | Keywords$ Haste | Triggers$ EOTSac | Duration$ Permanent | AtEOT$ Sacrifice | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/mizzix_replica_rider.txt:8]: AtEOT$ wrong

**Review by tool4ever — DISMISSED** [review]

**Comment by squee1968** [comment]
Not sure why my version failed, but thankee-sai for the assist.

**Review by squee1968 — COMMENTED** [review]
Not sure why my version failed, but thankee-sai for the assist.

File: forge-gui/res/cardsfolder/upcoming/mizzix_replica_rider.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+Name:Mizzix, Replica Rider
+ManaCost:4 R
+Types:Legendary Creature Goblin Wizard
+PT:4/5
+K:Flying
+T:Mode$ SpellCast | ValidCard$ Card.wasNotCastFromYourHand | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigCopy | TriggerDescription$ Whenever you cast a spell from anywhere other than your hand, you may pay {1}{U/R}. If you do, copy that spell and you may choose new targets for the copy. If the copy is a permanent spell, it gains haste and "At the beginning of the end step, sacrifice this permanent."
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/mizzix_replica_rider.txt:6]: Please add for card detail window: (A copy of a permanent spell becomes a token.)

**Review by paulsnoops — APPROVED** [review]

**Pull request merged**